QuizApp/App/Screens/Leaderboard/LeaderboardViewController.swift

```diff
@@ -0,0 +1,143 @@
+
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let view = tableView.dequeueReusableHeaderFooterView(withIdentifier: LeaderboardHeader.reuseIdentifier)
+        return view
+    }
+
+}
+
+extension LeaderboardViewController: UITableViewDelegate { }
+
+extension LeaderboardViewController: ConstructViewsProtocol {
+
+    func createViews() {
+        leaderboradTableView = UITableView()
+        view.addSubview(leaderboradTableView)
+    }
+
+    func styleViews() {
+        view.applyGradientWith([UIColor.loginBackgroundTop.cgColor, UIColor.loginBackgroundBottom.cgColor])
+
+        setupNavigationBar()
+        setupTableView()
+    }
+
+    func defineLayoutForViews() {
+        leaderboradTableView.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide).offset(10)
+            $0.leading.trailing.equalTo(view.safeAreaLayoutGuide)
+            $0.bottom.equalTo(view.safeAreaLayoutGuide).inset(30)
+        }
+    }
+
+}
+
+private extension LeaderboardViewController {
+
+    func addTableViewDelegateAndDatasource() {
+        leaderboradTableView.dataSource = self
+        leaderboradTableView.delegate = self
+    }
+
+    func addSubscription() {
+        viewModel
+            .$leaderboardEntries
+            .sink { [weak self] _ in
+                guard let self = self else { return }
+
+                self.leaderboradTableView.reloadData()
+
+            }
+            .store(in: &cancellables)
+    }
+
+    func setupNavigationBar() {
+        title = "Leaderboard"
+        navigationController?.navigationBar.titleTextAttributes = [
+            NSAttributedString.Key.foregroundColor: UIColor.white,
+            NSAttributedString.Key.font: Fonts.sourceSansProBold24.font]
+        navigationController?.navigationBar.tintColor = .white
+        navigationItem.setHidesBackButton(true, animated: false)
+
+        let closeBarButton = UIBarButtonItem(
+            image: UIImage.closeIcon,
+            style: .plain,
+            target: self,
+            action: #selector(closeTapped)
+        )
+        navigationItem.rightBarButtonItem = closeBarButton
+    }
+
+    @objc func closeTapped() {
+        viewModel.closeLeaderboard()
+    }
+
+    func setupTableView() {
+        leaderboradTableView.backgroundColor = .clear
+        leaderboradTableView.register(LeaderboardCell.self, forCellReuseIdentifier: LeaderboardCell.reuseIdentifier )
+        leaderboradTableView.register(
+            LeaderboardHeader.self,
+            forHeaderFooterViewReuseIdentifier: LeaderboardHeader.reuseIdentifier
+        )
+        leaderboradTableView.rowHeight = 63
+        leaderboradTableView.separatorInset = .init(top: 0, left: 0, bottom: 0, right: 0)
+        leaderboradTableView.separatorColor = .white
+        leaderboradTableView.estimatedSectionHeaderHeight = 50
```

Comment:
This is also `styleViews`.

Having function like `setupTableView` messes with the `ConstructViewsProtocol` in a way that code that should be in one of those methods is now extracted somewhere else and it's harder to find. Also, try to avoid using `setup` in method names because "setup" can mean anything and with time, `setup` functions tend to explode since anything is practically a setup (style, registration, defining constraints etc).